parsers/ELEXON.py

```diff
@@ -466,85 +382,49 @@ def fetch_exchange(
-    # modes at the moment. (16/12/2022) FUELINST will be decomissioned in 2023, so we should
-    # switch back to B1620 at some point.
-    response = query_production(session, target_datetime, "FUELINST")
-    fuel_inst_data = parse_production_FUELINST(response, target_datetime, logger)
-    raw_additional_data = query_additional_eso_data(target_datetime, session)
-    additional_data = parse_additional_eso_production(raw_additional_data)
-    data = process_production_events(fuel_inst_data, additional_data)
-    # We are fetching from FUELINST directly.
-    if False:
-        # At times B1620 has had poor quality data for wind so fetch from FUELINST
-        # But that source is unavailable prior to cutout date
-        HISTORICAL_WIND_CUTOUT = "2016-03-01"
-        FETCH_WIND_FROM_FUELINST = True
-        if target_datetime < arrow.get(HISTORICAL_WIND_CUTOUT).datetime:
-            FETCH_WIND_FROM_FUELINST = False
-        if FETCH_WIND_FROM_FUELINST:
-            wind = _fetch_wind(target_datetime, logger=logger)
-            for entry in data:
-                datetime = entry["datetime"]
-                wind_row = wind[wind["datetime"] == datetime]
-                if len(wind_row):
-                    entry["production"]["wind"] = wind_row.iloc[0]["Wind"]
-                else:
-                    entry["production"]["wind"] = None
-
-    required = ["coal", "gas", "nuclear", "wind"]
-    expected_range = {
-        # Historical data might be above the current capacity for coal
-        "coal": (0, 20000),
-        "gas": (100, 60000),
-        "nuclear": (100, 56000),
-        "wind": (0, 600000),
-    }
-    data = [
-        x
-        for x in data
-        if validate(x, logger, required=required, expected_range=expected_range)
-    ]
-
-    return data
-
-
-if __name__ == "__main__":
-    """Main method, never used by the Electricity Map backend, but handy for testing."""
-
-    print("fetch_production() ->")
-    print(fetch_production())
-
-    print("fetch_exchange(FR, GB) ->")
-    print(fetch_exchange("FR", "GB"))
-
-    print("fetch_exchange(GB, IE) ->")
-    print(fetch_exchange("GB", "IE"))
+    if target_datetime is None:
+        target_datetime = datetime.now(tz=timezone.utc)
+    else:
+        target_datetime = target_datetime.astimezone(timezone.utc)
 
-    print("fetch_exchange(GB, NL) ->")
-    print(fetch_exchange("GB", "NL"))
+    if target_datetime < ELEXON_START_DATE:
+        raise ParserException(
+            parser="ELEXON.py",
+            message=f"Production data is not available before {ELEXON_START_DATE.date()}",
+        )
 
-    print("fetch_exchange(GB, DK-DK1) ->")
-    print(fetch_exchange("DK-DK1", "GB"))
+    data = query_production(session, target_datetime, logger)
+    if not len(data):
+        data = query_and_merge_production_fuelhh_and_eso(
+            session, target_datetime, logger
+        )
```

Comment:
This does not work and break it all basically, first of all the first data always have a length cause it always have wind production but the rest of the values are 0.